api.php

```diff
@@ -59,34 +58,36 @@
             if (!is_array($array[0])) {
                 $array = [$array];
             }
+            $responseFalseId = [];
             $res = [];
             foreach ($array as $book) {
+                if (!is_integer($book['id']) or is_null($book['id'])) {
+                    $response = ['message' => 'no id, no changes', 'data' => $book, 'status' => 200];
+                    $responseFalseId = array_merge_recursive($responseFalseId, $response);
+                    continue;
+                }
                 $id = $book['id'];
-                unset($book['id']);
-                $result = update($book, $id);
-                array_push($res, $result);
+                $result = DB::getInstance()->update($book, $id);
+                $res = array_merge_recursive($res, $result);
             }
-            response($res);
+            $finishResponseArray = array_merge_recursive($res, $responseFalseId);
+            response($finishResponseArray);
         }
         if (preg_match("[^/api/books/([0-9]{1,})(|/)$]", $_SERVER['REQUEST_URI'])) {
-            preg_match_all('/[0-9]/', $_SERVER['REQUEST_URI'], $matches);
-            $id = implode($matches[0]);
-            if (!is_array($array[0])) {
-                $array = [$array];
+            if (is_array($array[0])) {
+                response(['message' => 'Incorrect input data', 'data' => $array, 'status' => 200]);
             }
-            $res = [];
-            foreach ($array as $book) {
-                $result = update($book, $id);
-                array_push($res, $result);
-            }
-            response($res);
+            preg_match('/[0-9]+/', $_SERVER['REQUEST_URI'], $matches);
+            $id = implode($matches);
+            $result = DB::getInstance()->update($array, $id);
+            response($result);
         }
         response(['message' => 'incorrect URL', 'data' => 'error', 'status' => 404]);
     case "DELETE":
         if (preg_match("[^/api/books/([0-9]{1,})(|/)$]", $_SERVER['REQUEST_URI'])) {
             preg_match_all('/[0-9]/', $_SERVER['REQUEST_URI'], $matches);
             $id = implode($matches[0]);
-            $result = del($id);
+            $result = DB::getInstance()->del($id);
             response($result);
         }
         response(['message' => 'incorrect URL', 'data' => 'error', 'status' => 404]);

```

Comment:
This command can be after whole block of switch-case one time because if method + url are correct response will be correct.